### PR TITLE
Add Nginx front-end with Let's Encrypt SSL

### PR DESCRIPTION
**Pull request by tgross**
For https://github.com/autopilotpattern/jenkins/issues/6

@misterbisson @jasonpincin this puts Nginx in front of Jenkins, using the same Let's Encrypt setup that we have in autopilotpattern/nginx. The only thing we've changed from the Nginx config is adding the recommended reverse proxy for Jenkins.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -104,7 +104,7 @@ check() {
     echo 'GITHUB_JOBS_SPEC=' >> _env
     echo >> _env
 
-    echo CONSUL=consul.svc.${TRITON_ACCOUNT}.${TRITON_DC}.cns.joyent.com >> _env
+    echo CONSUL=jenkins-consul.svc.${TRITON_ACCOUNT}.${TRITON_DC}.cns.joyent.com >> _env
```
Comment (misterbisson) [inline setup.sh:107]: Thank you for namespacing this.

**Comment by misterbisson** [comment]
🏡 🚶

**Comment by charandas** [comment]
@tgross Great work on this. Someone probably forgot to push: `autopilotpattern/jenkins-nginx:latest`

I can locally build for now from [here](https://github.com/autopilotpattern/jenkins/tree/master/nginx).

**Comment by tgross** [comment]
@charandas good catch. Done and tagged as https://github.com/autopilotpattern/jenkins/releases/tag/2.19r1.1 and on the Docker Hub as `autopilotpattern/jenkins:2.19r1.1` and `autopilotpattern/jenkins-nginx:2.19r1.1` (and `:latest` for both).

**Pull request merged**